File: Le0sy/step09/_4948.java
package step9;

import java.util.Scanner;

public class _4948 {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        boolean[] chkPrime = new boolean[246913];
        chkPrime[0] = chkPrime[1] = true;

        for (int i = 2; i <= Math.sqrt(chkPrime.length); i++) {
            if (chkPrime[i]) continue;
            for (int j = i * i; j < chkPrime.length; j += i) {
                chkPrime[j] = true;
            }
        }
        int input;
        while ((input = sc.nextInt()) != 0) {
            int cnt = 0;
            for (int i = input + 1; i <= input * 2; i++) {
                if (!chkPrime[i]) {
                    cnt++;
                }
            }
            System.out.println(cnt);
        }
    }
}

File: hysKim1/4948.py
while True:
    n = int(input())
    if n == 0:
        break
    n1, n2= n+1, 2*n
    li = [1]*(n2+1)
    for i in range(2, int((n2+1)**0.5) + 1):
        if li[i] == 1:
            for j in range(i+i, n2+1, i):
                li[j] = 0
    cnt = 0
    for i in range(n1, n2+1):
        if li[i] == 1 and i > 1:
            cnt += 1
    print(cnt)
